src/coreclr/jit/scev.cpp

```diff
@@ -471,34 +469,34 @@ Scev* ScalarEvolutionContext::AnalyzeNew(BasicBlock* block, GenTree* tree, int d
 
             assert(ssaDsc->GetBlock() != nullptr);
 
-            // We currently do not handle complicated addrecs. We can do this
-            // by inserting a symbolic node in the cache and analyzing while it
-            // is part of the cache. It would allow us to model
-            //
-            //   int i = 0;
-            //   while (i < n)
-            //   {
-            //     int j = i + 1;
-            //     ...
-            //     i = j;
-            //   }
-            // => <L, 0, 1>
-            //
-            // and chains of recurrences, such as
-            //
-            //   int i = 0;
-            //   int j = 0;
-            //   while (i < n)
-            //   {
-            //     j++;
-            //     i += j;
-            //   }
-            // => <L, 0, <L, 1, 1>>
-            //
-            // The main issue is that it requires cache invalidation afterwards
-            // and turning the recursive result into an addrec.
-            //
-            return CreateSimpleAddRec(store, enterScev, ssaDsc->GetBlock(), ssaDsc->GetDefNode()->Data());
+            Scev* simpleAddRec = CreateSimpleAddRec(store, enterScev, ssaDsc->GetBlock(), ssaDsc->GetDefNode()->Data());
```

Comment:
It might be useful to retain the prior comment here and augment so there's as a worked-out example of how these symbolic addrecs end up getting resolved.

Reply:
Makes sense... let me add it as part of a future PR.